src/components/table/columns.ts

```diff
@@ -181,13 +182,37 @@ export function createCustomComponent(
     };
 
     element.style.display = 'inline-block';
-    Object.assign(element, props);
+    setElementProperties(element, props);
 
     createResizeObserver(element, cell.getColumn());
 
     return element;
 }
 
+function setElementProperties(element: HTMLElement, props: object) {
+    const properties = pickBy(props, negate(isEventListener));
+    Object.assign(element, properties);
+
+    const listeners = pickBy(props, isEventListener);
+    Object.entries(listeners).forEach(([key, value]) => {
+        const event = getEventName(key);
+        element.addEventListener(event, value as any);
+    });
+}
+
+function isEventListener(value: any, key: string): boolean {
+    if (typeof value !== 'function') {
+        return false;
+    }
+
+    return /^on[A-Z]/.test(key);
+}
+
+function getEventName(propertyName: string): string {
+    // eslint-disable-next-line no-magic-numbers
+    return propertyName.charAt(2).toLowerCase() + propertyName.slice(3);
```

Comment:
I like the solution but I suggest that we add some kind of comment or docstring that explains what's going on. It took quite a bit of time for me to realise what we're doing :)

Reply:
7a9f5ff